lib/serialport.js

```diff
@@ -4,6 +4,7 @@
 
 // 3rd Party Dependencies
 var debug = require('debug')('serialport');
+var escStr = require('lodash').escape;
```

Comment:
Don't use escape function from lodash because it doesn't escape \r and \n symbols. You can create own function:
function escStr(data) {
  return data ? data.toString().replace(/\n/g, "\\n").replace(/\r/g, "\\r") : '';
}